src/flamegraph/svg.rs

```diff
@@ -48,224 +48,270 @@ pub(super) struct StyleOptions<'a> {
+            "flamegraph.js"
+        ))))?;
+        svg.write_event(Event::End(BytesEnd::borrowed(b"script")))?;
 
-    write_str(
-        svg,
-        &mut buf,
-        TextItem {
-            color: "rgb(0, 0, 0)",
-            size: super::FONTSIZE,
-            x: super::XPAD as f64,
-            y: (style_options.imageheight - (super::YPAD2 / 2)) as f64,
-            text: " ".into(),
-            location: None,
-            extra: iter::once(("id", "details")),
-        },
-    )?;
+        svg.write_event(Event::Empty(
+            BytesStart::borrowed_name(b"rect").with_attributes(vec![
+                ("x", "0"),
+                ("y", "0"),
+                ("width", &*format!("{}", self.image_width)),
+                ("height", &*format!("{}", style_options.imageheight)),
+                ("fill", "url(#background)"),
+            ]),
+        ))?;
 
-    write_str(
-        svg,
-        &mut buf,
-        TextItem {
-            color: "rgb(0, 0, 0)",
-            size: super::FONTSIZE,
-            x: super::XPAD as f64,
-            y: (super::FONTSIZE * 2) as f64,
-            text: "Reset Zoom".into(),
-            location: None,
-            extra: vec![
-                ("id", "unzoom"),
-                ("onclick", "unzoom()"),
-                ("style", "opacity:0.0;cursor:pointer"),
-            ],
-        },
-    )?;
+        // We don't care too much about allocating just for the prelude
+        let mut buf = StrStack::new();
+        self.write_str(
+            svg,
+            &mut buf,
+            TextItem {
+                color: "rgb(0, 0, 0)",
+                size: self.font_size + 5,
+                x: (self.image_width / 2) as f64,
+                y: (self.font_size * 2) as f64,
+                text: self.title.clone().into(),
+                location: Some("middle"),
+                extra: None,
+            },
+        )?;
 
-    write_str(
-        svg,
-        &mut buf,
-        TextItem {
-            color: "rgb(0, 0, 0)",
-            size: super::FONTSIZE,
-            x: (super::IMAGEWIDTH - super::XPAD - 100) as f64,
-            y: (super::FONTSIZE * 2) as f64,
-            text: "Search".into(),
-            location: None,
-            extra: vec![
-                ("id", "search"),
-                ("onmouseover", "searchover()"),
-                ("onmouseout", "searchout()"),
-                ("onclick", "search_prompt()"),
-                ("style", "opacity:0.1;cursor:pointer"),
-            ],
-        },
-    )?;
+        match self.subtitle.clone() {
+            Some(subtitle) => self.write_str(
+                svg,
+                &mut buf,
+                TextItem {
+                    color: "rgb(160, 160, 160)",
+                    size: self.font_size,
+                    x: (self.image_width / 2) as f64,
+                    y: (self.font_size * 4) as f64,
+                    text: subtitle.into(),
```

Comment:
I don't think you need the `.into()` here

Reply:
Unfortunately it does not work either:
```
error[E0308]: mismatched types
   --> src/flamegraph/svg.rs:195:27
    |
195 |                     text: subtitle,
    |                           ^^^^^^^^ expected enum `flamegraph::svg::TextArgument`, found reference
    |
    = note: expected type `flamegraph::svg::TextArgument<'_>`
               found type `&std::string::String`
```

Reply:
Ah, yes, you might still need the `.into()` to make a `TextArgument` out of the `&str`, you're right!